### PR TITLE
Improve `hf gallagher` commands

### PR DESCRIPTION
**Pull request by DarkMatterMatt**
When creating/deleting applications:
- Allow skipping Card Application Directory (CAD) updates
- Allow skipping application creation/deletion (only updates the CAD)
- Allow specifying the Gallagher CAD key (useful when using the same card with multiple site keys)
- Use GetAIDs to find an available Application ID (AID), was previously using trial&error to find an unused AID

Other changes:
- Continue reading credentials if one fails (e.g. when a card is used with multiple site keys)
- Remove incorrect AID validation (testing suggests that they are not limited as previously suggested)
- Update changelog (oops, forgot to do that last time 😞)

### CODE REVIEW

File: client/src/cmdhfgallagher.c [hunk 1]
```diff
@@ -865,89 +933,93 @@ static int CmdGallagherClone(const char *cmd) {
...
                  );
 
     void *argtable[] = {
         arg_param_begin,
-        arg_int0("n",  "keynum",   "<dec>", "Key number [default = 0]"),
-        arg_str0("t",  "algo",     "<DES|2TDEA|3TDEA|AES>", "Crypt algo: DES, 2TDEA, 3TDEA, AES"),
-        arg_str0("k",  "key",      "<hex>", "Key for authentication to the PICC"),
+        arg_int0("n",   "keynum",  "<dec>", "PICC key number [default = 0]"),
+        arg_str0("t",   "algo",    "<DES|2TDEA|3TDEA|AES>", "PICC crypt algo: DES, 2TDEA, 3TDEA, AES"),
+        arg_str0("k",   "key",     "<hex>", "Key for authentication to the PICC to create applications (HEX 8(DES), 16(2TDEA or AES) or 24(3TDEA) bytes)"),
```
Comment (iceman1001) [inline client/src/cmdhfgallagher.c:944]: we have limited screen space,  so long help-text for parameter is discouraged.  Instead write in the command help description if you need to make something clear for a param

**Comment by iceman1001** [comment]
Nice improvements,
and yes, you forgot last time to add a line in changelog.  No hard done though. 

Two things comes to mind,  we have limited screen space.  
so long param names or param help text is discouraged.   

Otherwise I will merge this one

File: CHANGELOG.md [hunk 2]
```diff
@@ -62,6 +63,7 @@ This project uses the changelog in accordance with [keepchangelog](http://keepac
  - Fixed `SimulateTagLowFrequencyEx` ignoring the `ledcontrol` argument (@zabszk)
  - Added ledcontrol argument to LF operations (@zabszk)
  - Added new standalone mode `dankarmulti` -  select and run multiple standalone modes (@dankar)
+ - Added `hf gallagher` commands for read/writing DESFire cards (@DarkMatterMatt)
```
Comment (doegox) [inline CHANGELOG.md:66]: You can move this line up between current lines 9 & 10 (above " - Added support for MIFARE DESFire application creation without authentication ")

**Comment by iceman1001** [comment]
Thanks!

**Pull request merged**